app/components/AddCompanyPopUp.js

```diff
@@ -238,16 +279,37 @@ export default class AddCompanyPopUp extends Component {
                 open={this.state.open}
                 onRequestClose={() => this.handleClose()}
                 autoScrollBodyContent={true}>
-            <TextField floatingLabelText='Company Name'
-                       hintText='Ex: ABC Co.'
-                       onChange={(e) => this.setState({ name: e.target.value })}></TextField>
-            <TextField floatingLabelText='City'
-                       hintText='Ex: Denver'
-                       onChange={(e) => this.setState({ city: e.target.value })}></TextField>
+             <AutoComplete
+                 floatingLabelText="Company Name"
+                 hintText='Ex. ABC Co.'
+                 maxSearchResults={4}
+                 onNewRequest={(name) => this.setState({ name })}
+                 filter={AutoComplete.fuzzyFilter}
+                 dataSource={this.state.allCompanies}
+                 onUpdateInput={(e) => this.setState({ name: e }) }
+                 openOnFocus
+               />
+             <AutoComplete
+                 floatingLabelText="City"
+                 hintText='Ex. Denver'
+                 maxSearchResults={4}
+                 onNewRequest={(city) => this.setState({ city })}
+                 filter={AutoComplete.fuzzyFilter}
+                 dataSource={this.state.allCities}
+                 onUpdateInput={(e) => this.setState({ city: e }) }
+                 openOnFocus
```

Comment:
Can this be changed to a keyup or something? Just some kind of check to see if the company is already in the db.